### PR TITLE
14120 converter v3

### PR DESCRIPTION
**Pull request by sergei-bronnikov**
https://codio.myjetbrains.com/youtrack/issue/codio-14120

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -xe
+
+commit=$1
+out_dir=$2
+
+output_name="guides-converter-v3-$commit"
+go build -ldflags "-s -w" -o $out_dir/$output_name ./cmd/guides-converter-v3
```
Comment (destitutus) [inline build.sh]: surround the variable with double quotes`"$out_dir/$output_name"`

File: internal/app/app.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		return fmt.Errorf("content convert error")
+	}
+	if err := cleanup.AfterSuccessfull(); err != nil {
+		return fmt.Errorf("cleanup error")
+	}
+	return nil
+}
+
+func alreadyInProgress() (bool, error) {
+	if _, err := os.Stat(constants.AlreadyInProgressFlag); err == nil {
```
Comment (destitutus) [inline internal/app/app.go]: I am not sure if it good solution for lock a process - a file will not be cleaned up if box will be restarted during a process
Probably, it is better to use `syscall.Flock` call

File: internal/utils/utils.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		RemoveDirectory(dir)
+	}
+	return nil
+}
+
+func MakeDir(directory string) error {
+	if stat, err := os.Stat(directory); err == nil && stat.IsDir() {
+		return nil
+	}
+	if err := os.Mkdir(directory, 0777); err != nil {
```
Comment (destitutus) [inline internal/utils/utils.go]: `777` is to wide` 755` should be enough I think

File: internal/utils/utils.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+		return err
+	}
+	if err := json.Unmarshal(bytes, parsed); err != nil {
+		return err
+	}
+	return nil
+}
+
+func WriteJson(path string, content map[string]interface{}) error {
+	jsonFile, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0666)
```
Comment (destitutus) [inline internal/utils/utils.go]: `644`

**Review by destitutus — APPROVED** [review]

File: runConvert.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
```
Comment (destitutus) [inline runConvert.sh:2]: `set -xe`

File: runConvert.sh [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  "$@"
+  ret=$?
+  if [[ $ret -ne 0 ]]
+  then
+      rm guides-converter-v3
+      exit $ret
+  fi
+}
+
+curl "https://static-assets.codio.com/guides-converter-v3/guides-converter-v3-${converterVersion}" --output guides-converter-v3
```
Comment (destitutus) [inline runConvert.sh]: `run_with_failover`

**Review by destitutus — DISMISSED** [review]

**Review by destitutus — DISMISSED** [review]

**Review by destitutus — DISMISSED** [review]

**Review by mint87 — APPROVED** [review]

**Pull request merged**